clients/src/main/java/org/apache/kafka/common/requests/OffsetFetchRequest.java

```diff
@@ -174,6 +315,28 @@ public boolean isAllPartitions() {
         return data.topics() == ALL_TOPIC_PARTITIONS;
     }
 
+    public boolean isAllPartitionsForGroup(String groupId) {
+        OffsetFetchRequestGroup group = data
+            .groupIds()
+            .stream()
+            .filter(g -> g.groupId().equals(groupId))
+            .collect(toSingleton());
+        return group.topics() == ALL_TOPIC_PARTITIONS_BATCH;
+    }
+
+    // Custom collector to filter a single element
+    private <T> Collector<T, ?, T> toSingleton() {
```

Comment:
We seem to have two copies of this method in the PR. We need to decide first whether this logic is correct first. If we really do want the method, we need to make it shared.

Reply:
Makes sense, for now I will remove iit and convert the methods that used this to collect a `List` and get the first element in the list.

Reply:
Can we a add a test in the new `kafka.server.OffsetFetchRequestTest`? Can do that in a separate PR.

Reply:
Would this be a test for single group requests with v8? I have added that case to the existing single group tests for older versions in `OffsetFetchRequestTest.scala` and will push those changes shortly. Is there another type of test you were thinking of?

Reply:
Same test class `OffsetFetchRequestTest.scala`, a test with v8 with a batched request where the same group appears twice, perhaps with different topics. The response should be either InvalidRequestException because we want to treat it as an error OR actual offsets because we handle the request correctly.

Reply:
Ah okay, good point. Let's do this in a follow up PR, since we are trying to get this feature into 3.0 right now.

Reply:
@rajinisivaram I'm not sure how we could test that case, as the `Builder` for the `OffsetFetchRequest` takes in a `Map<String, List<TopicPartition>>` which means all of the keys have to be unique. As a result if a group appears twice, the latest entry will take precedence. Not sure if this is how we want to handle it, but generally speaking I think we don't support having the same name group id appearing twice in the request - as in it's not possible to construct a request like that, at least not using the `Builder`

Reply:
You can probably set the list in `OffsetFetchRequestData` directly in the test. But let's leave that for a follow-on PR.

Reply:
Opened https://issues.apache.org/jira/browse/KAFKA-13045 to track this.

Reply:
Opened https://github.com/apache/kafka/pull/11000 for this. We can continue the discussion there.